pkg/azure/blob_test.go

```diff
@@ -115,6 +115,7 @@ func TestNewClientAndBucketExistsWithProxy(t *testing.T) {
 
 			ok, err := client.BucketExists(context.Background(), "podinfo")
 			if tt.err != "" {
+				g.Expect(err).To(HaveOccurred())
 				g.Expect(err.Error()).To(ContainSubstring(tt.err))
```

Comment:
This approach still causes panic because the second assertion `err.Error()` executes even after the first assertion fails.

Consider using `g.Expect(err).To(MatchError(ContainSubstring(tt.err)))` instead, which safely handles both the nil check and substring matching in a single assertion.

```suggestion
                                g.Expect(err).To(MatchError(ContainSubstring(tt.err)))
```